### DISCUSSION TITLE
AWS Async APIs too slow for first run

### DISCUSSION THREAD

**Discussion by ksvasan** [aws/aws-sdk-java-v2#3159] [General]
We're using the below AWS APIs extensively for many of our customers on the Java SDK.
1. EC2 - describe-flow-logs
2. Logs - describe-log-streams

The issue is that we're seeing that these calls take a lot of time when coupled with [reactor](https://projectreactor.io/) (but only for the first run). We're using the below APIs (a lot of times with former and latter throttled to 2.5 rps)
1. EC2 DescribeFlowLogsPaginator API (on Ec2AsyncClient)
2. CWLogs DescribeLogStreamsPaginator API (on CloudWatchLogsAsyncClient)

Even when I limit the reactive pipeline to stop with (1) it takes lots of time to list all the flowlogs across multiple customers. Is there a reason for this behavior (is this default for reactor - could it be something like [this](https://stackoverflow.com/questions/68457545/project-reactor-publishers-initialisation-taking-time-for-first-time-initialisat)?)

To give a sample of the delays that we're seeing a normal load pipeline running in 10 hrs takes like 60-80 hrs in the first run, subsequent runs are close to the 10 hr mark. Is there any reason for this behavior, or have we seen this happening before, or is there a way to speed this up?

AWS Java SDK: 2.17.116
Reactor: 3.4.13

**Comment by debora-ito** [comment]
@ksvasan I'm not familiar with how Reactor works, never run the SDK coupled with Reactor before so don't really know the answer... 
Can you provide a self-contained sample we can run to test it?

Maybe someone in the community can help.
